core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -147,21 +185,69 @@ static Snapshot fromJson(FileIO io, JsonNode node) {
 
     Integer schemaId = JsonUtil.getIntOrNull(SCHEMA_ID, node);
 
+    List<StatisticsFile> statistics = null;
```

Comment:
not sure how this project deals with null, how about init this to empty collection and void null checking as possible?

Reply:
@zinking i agree with you and what you suggest is how i write code myself. However, i think i am following project rules here.